src/utils/onlineParser.js

```diff
@@ -13,62 +13,47 @@
  *
  * Options:
  *
+ *     eatWhitespace: (stream: Stream | GraphQLStream) => boolean
+ *       Use CodeMirror API.
+ *
  *     LexRules: { [name: string]: RegExp }, Includes `Punctuation`.
  *
  *     ParseRules: { [name: string]: Array<Rule> }, Includes `Document`.
  *
- *     eatWhitespace: (stream: CodeMirrorStream) => boolean, Use CodeMirror API.
+ *     editorConfig: { [name: string]: mixed }, Provides an editor-specific
+ *       configurations set.
  *
  */
+
+import GraphQLStream from './GraphQLStream';
+
 export default function onlineParser(options) {
-  const { eatWhitespace, LexRules, ParseRules } = options;
   return {
     startState() {
       var initialState = { level: 0 };
-      pushRule(ParseRules, initialState, 'Document');
+      pushRule(options.ParseRules, initialState, 'Document');
       return initialState;
     },
-    getToken(stream, state) {
-      return getToken(eatWhitespace, LexRules, ParseRules, this, stream, state);
+    token(stream, state) {
+      return getToken(stream, state, options);
+    },
+    runParser(sourceText, callbackFn) {
+      const state = this.startState();
+      const lines = sourceText.split('\n');
```

Comment:
I'm not sure. I think there are pros/cons to splitting lines - pros being it's easier to calculate stream positions, and cons including code not being as clean as it could be. Maybe `runParser` would help to facilitate.